tests/test_multitiffmultipageextractor.py

```diff
@@ -0,0 +1,266 @@
+    scanimage_extractor = ScanImageImagingExtractor(
+        file_paths=file_paths,
+        channel_name="Channel 1",
+    )
+
+    # Initialize MultiTIFFMultiPageExtractor with equivalent mapping
+    multi_extractor = MultiTIFFMultiPageExtractor(
+        file_paths=file_paths,
+        sampling_frequency=scanimage_extractor.get_sampling_frequency(),
+        dimension_order="CZT",  # ScanImage uses this order
+        num_channels=2,
+        channel_index=0,
+        num_planes=1,  # For this example, num_planes is 1
+    )
+
+    # Compare basic properties
+    assert multi_extractor.get_num_samples() == scanimage_extractor.get_num_samples()
+
+    # Compare frame shapes - ScanImage includes depth dimension, MultiTIFF doesn't
+    scanimage_shape = scanimage_extractor.get_frame_shape()
+    multi_shape = multi_extractor.get_frame_shape()
+
+    # Check that the first two dimensions (height, width) match
+    assert multi_shape[0] == scanimage_shape[0]
+    assert multi_shape[1] == scanimage_shape[1]
+
+    assert multi_extractor.get_sampling_frequency() == scanimage_extractor.get_sampling_frequency()
+
+    # Compare the actual series data
+    scanimage_series = scanimage_extractor.get_series().squeeze()
+    multi_series = multi_extractor.get_series()
+
+    assert_allclose(scanimage_series, multi_series, rtol=1e-5, atol=1e-8)
+
+
+def test_channel_extraction(tmp_path):
+    """Test extraction of specific channels."""
+    # Create multi-channel test data
+    tifffile = get_package(package_name="tifffile")
+
+    # Create a file with multiple channels
+    num_channels = 2
+    num_planes = 2
+    num_timepoints = 3
+    height = 8
+    width = 10
+
+    file_path = tmp_path / "multichannel.tif"
+    pages = []
+    for t in range(num_timepoints):
+        for c in range(num_channels):
+            for z in range(num_planes):
+                page_data = np.ones((height, width), dtype=np.uint16) * (t * 100 + c * 10 + z + 1)
+                pages.append(page_data)
+
+    tifffile.imwrite(file_path, pages)
+
+    # Test extracting first channel
+    extractor_ch0 = MultiTIFFMultiPageExtractor(
+        file_paths=[file_path],
+        sampling_frequency=30.0,
+        dimension_order="TCZ",
+        num_channels=num_channels,
+        channel_index=0,
+        num_planes=num_planes,
+    )
+
+    # Test extracting second channel
+    extractor_ch1 = MultiTIFFMultiPageExtractor(
+        file_paths=[file_path],
+        sampling_frequency=30.0,
+        dimension_order="TCZ",
+        num_channels=num_channels,
+        channel_index=1,
+        num_planes=num_planes,
+    )
+
+    assert extractor_ch0.get_num_samples() == num_timepoints
+    assert extractor_ch1.get_num_samples() == num_timepoints
+
+    # Verify different channels have different data
+    series_ch0 = extractor_ch0.get_series()
+    series_ch1 = extractor_ch1.get_series()
+
+    assert not np.array_equal(series_ch0, series_ch1)
+
+
+def test_error_handling(tmp_path):
```

Comment:
I can't see the code coverage report for some reason. But I believe there's quite a few error cases that should probably be tested here, like the RuntimeError and the ValueError on invalid dimension sizes, etc. Also, I don't think there's any reason to bundle them up into one test. We can probably split them out. So test_invalid_file_path, test_invalid_dimension_order, test_invalid_tiff_file, and so on.